docs/testing-guide/avm-types.md

```diff
@@ -0,0 +1,246 @@
+# AVM Types
+
+These types are available directly under the `algopy` namespace. They represent the basic AVM primitive types and can be instantiated as follows:
+
+## UInt64
+
+```python
+# Direct instantiation
+uint64_value = algopy.UInt64(100)
+
+# Instantiate test context
+...
+
+# Generate a random UInt64 value
+random_uint64 = ctx.any_uint64()
+
+# Specify a range
+random_uint64 = ctx.any_uint64(min_value=1000, max_value=9999)
+```
+
+## Bytes
+
+```python
+# Direct instantiation
+bytes_value = algopy.Bytes(b"Hello, Algorand!")
+
+# Instantiate test context
+...
+
+# Generate random byte sequences
+random_bytes = ctx.any_bytes()
+
+# Specify the length
+random_bytes = ctx.any_bytes(length=32)
+```
+
+## String
+
+```python
+# Direct instantiation
+string_value = algopy.String("Hello, Algorand!")
+
+# Instantiate test context
+...
+
+# Generate random strings
+random_string = ctx.any_string()
+
+# Specify the length
+random_string = ctx.any_string(length=16)
+```
+
+## BigUInt
+
+```python
+# Direct instantiation
+biguint_value = algopy.BigUInt(100)
+
+# Instantiate test context
+...
+
+# Generate a random BigUInt value
+random_biguint128 = ctx.any_biguint128()
+
+# Specify a range for UInt128
+random_biguint128 = ctx.any_biguint128(min_value=1000, max_value=999999999999)
+
+# Generate a random UInt256 value
+random_biguint256 = ctx.any_biguint256()
+
+# Specify a range for UInt256
+random_biguint256 = ctx.any_biguint256(min_value=1000, max_value=999999999999)
+
+# Generate a random UInt512 value
+random_biguint512 = ctx.any_biguint512()
+
+# Specify a range for UInt512
+random_biguint512 = ctx.any_biguint512(min_value=1000, max_value=999999999999)
+```
+
+## Asset
+
+```python
+# Direct instantiation
+asset = algopy.Asset(id=1, name="TestCoin", total=1000000, decimals=6)
```

Comment:
Remove property definitions from direct instantiation for reference types (asset, account, application)